### ISSUE TITLE
Lua error right after the intro

### ISSUE THREAD

**Issue by AurelioMV** [PhantasarProductions/TFTRevamp#602]
![image](https://user-images.githubusercontent.com/38708770/39249840-eb8717f0-4875-11e8-8726-fea358076fbd.png)

I watched the intro, chose to play with Jake, watched 2 more dialog screens and when it seemed the game itself was about to begin I got this error. Also the frame rate was incredible slow, seemed to be below 10 FPS. I was playing with a 1176x664 32 bit resolution.

**Comment by AurelioMV** [comment]
UPDATE: Restarted the game using the 800x600 16 bit resolution and it worked fine. However, the FPS is still an issue.

**Comment by Tricky1975** [comment]
I shall investigate this, as that error appears pretty odd to me to be related to a screen resolution. I do know some crashes do appear randomly, and are therefore hard to replicate. It does look like something went wrong during the initiation of the character data of Jake judging by the data shown in the screenshot. Thanks a lot for that screenshot ;)

The frame rate is a known issue, and hard to "fix", as the game has a pretty demanding engine and when other programs are running at the same time it slows down terribly. Webbrowsers, e-mail clients and torrent clients in particular can cause some trouble here, in my experience. The game was written on a Mac, so Windows behavior has always been hard to check for me... Of course, I don't know how fast your hardware is compared to mine, but I can say that my mac is a bit outdated and one of the cheaper models from the start too ;) 

The BlitzMax programming language (in which the underlying engine was written) did show me some issues before when a lot of graphics objects are being processed and that is exactly what is happening in the game every frame, and the engine owes a lot of its versatility which has been used in many locations of the game, but it can sometimes be a bit of a killer, particularly on slower computers. A lot of enhancements were already done, as the first experimental versions of this engine even before any playable content was written were even slower. XD
10 FPS appears very very slow though, although the game does skip a few frames from changing stuff in order not to animate the sprites too fast (so it is possible the true FPS is a little higher than 10). The approach was taken to make other graphic effects possible in higher speed. If there's something I can find out to increase the speed I will surely test it out and put it in the game.

Now the towns are known as problem children more than the dungeons and battle screens. The towns just contain more detail. I may need to put in some possibility to turn off certain graphics effects produced later in the game though for people using a slower computer :-/

**Comment by AurelioMV** [comment]
PS: I tried running as an administrator and with compatibility
configurations, such as reduced color scheme, windows XP mode, etc... But
still no difference.
PS2: I'm using a Windows 7 64 bits.

2018-04-25 19:05 GMT-03:00 Jose Aurelio <aureliomv93@gmail.com>:

**Comment by AurelioMV** [comment]
Maybe it wasn't screen related after all. Could very well be a coincidence
as I only tried the higher resolution once. About the hardware, I have a
pretty decent computer, it's a i7-7700k, 16 GB RAM, SSD, GTX 1070. Maybe
it's because of the hyperthreading, it's known that some older games don't
do well with it. I attached a video showing the performance, it's hard to
get past through the first 2 speeches. After the game itself begins it
takes forever to move a short distance, seems to be less than a frame per
second actually.

Thanks for responding, I really would like to play, I'm a big fan of RPGs.

2018-04-25 18:26 GMT-03:00 Jeroen Broks <notifications@github.com>:

**Comment by AurelioMV** [comment]
I checked the CPU usage, it's completely taking up one core. The other
cores are idle. But even a single core is working at 4.2GHz. Should be
enough to run it smoothly.

2018-04-25 19:14 GMT-03:00 Jose Aurelio <aureliomv93@gmail.com>:

**Comment by Tricky1975** [comment]
"it's known that some older games don't do well with it"

It would explain why, my 63 fires of lung project gives me the feeling to do this faster even though the underlying map engine was a translation from BlitzMax (which compiles into machine language) to Lua (which is 100% interpreted) which you'd say that would actually make things slower (63 fires is being written with LÖVE, btw), as I did hear that the BlitzMax engine's graphic module in particular was showing a bit with age. 
Let compare your hardware to mine, shall we?
![image](https://user-images.githubusercontent.com/11202073/39276638-44f33188-48ea-11e8-9fd0-6c2ac8243df6.png)

Hmm, looks like your computer outranks mine. Now I have seen Windows users with lower hardware not suffering in this department, so I really wonder what is "wrong" in your computer causing this slowdown to happen. Actually I guess the game should run faster on your computer than mine. Of course comparing Mac to Windows is always an unwise thing to do, due to some different approach to matters, but still... 

Now I guess this issue tracker did not send that video of yours through, but I'll take your word for it. Even with the differences between Mac and Windows, I deem it very odd that this happens to you considering your hardware, and I'm afraid this goes into a field in which I am no expert. I shall have to find some people who have an idea what may cause this... :-/

My apologies for your inconvenience, as I feel very bad about this situation... :cry:

**Comment by AurelioMV** [comment]
No problem, I know how hard it is to support different plataforms and
stuff. I'll wait for the 63 Fires to come out! :D

2018-04-25 19:49 GMT-03:00 Jeroen Broks <notifications@github.com>:

**Comment by Tricky1975** [comment]
I'll let you know when the preview-demo is done, but I must say that will take some months at least. If 63 works indeed faster, then it's good to know, as I was planning a migration operation for my 3 older RPG games to the new engine (also because LÖVE is more friendly on Linux that BlitzMax is)...
I'll link a friend of mine to this issue. Perhaps he knows how this can happen.

**Comment by Tricky1975** [comment]
My friend responded with this:
~~~
windows toets + G zodat hij zn systeem in game modus zet scheelt veel meer dan een lagere resolutie
maar dat is pas vanaf windows 8
meer dan voldoende ram
~~~
Of course, I take it you don't speak Dutch, not to mention he spoke in a bit of what we call "telegram style", so I'll translate it for you:
- Windows key + G so he'll set his system into game mode will do a lot more than a lower resolution
- That is only since Windows 8
- RAM is more than enough.

I don't know if this is helpful for you. I'm just passing on his analysis report ;)

**Comment by AurelioMV** [comment]
Thanks for asking you friend. But I'm using windows 7. I don't like windows
8 and windows 10 new style.

Em qui, 26 de abr de 2018 09:38, Jeroen Broks <notifications@github.com>
escreveu:

**Comment by Tricky1975** [comment]
Yeah, can't say you're wrong. Now I've been out of Windows since Windows Vista (as I switched to Mac somewhere near the end of the XP time and the start of the Vista time), but at my work places I've used both 7 and 10, and I must say that XP and 7 were the least terrible Windows instalments I've worked with and 10 has some features I can't turn off that I even deem unacceptable (auto-updates rebooting my computer without asking. I've already lost some data because of that). The point is of course, that if the support for Win7 isn't already deprecated, it'll happen soon, but of course, that's not my concern. I do agree that 7 is better than 10 (and even more better than 8). 
People can say about macs all they want, but Apple has been consistent with the Mac UI-setup for quite a long while.... Unlike Microsoft.

I'm not sure if Windows 7 is the true problem, though. Star Story was released before Windows 10 came (I think) and I never had complaints about FPS there unless people really had bad hardware. Star Story uses (an older version of) the same engine I used for TFT-REVAMPED. Technically even XP should be able to run the game (although it may cause some problems), any version before that, most likely not...

What I wonder is, my game "The Secrets of Dyrt" (also RPG) uses the same map engine as "Cynthia Johnson" (Puzzle game) does, although the latter uses a Lua-port version of that map engine in Lua. Of course, honesty requires me to say that Dyrt uses it in a bigger scale, the game is sophisticated enough to ignore what falls outside the screen range. Can you tell me if you have big differences in FPS or other performance issues between those two games? (Fun fact: The story of Dyrt takes place 200 years after the events in The Fairy Tale, however there are no real spoilers between the two games, so it doesn't matter which game you play first).
The reason I ask is because the Dyrt engine is also written in BlitzMax like this game, but the engine Cynthia Johnson uses, is the same engine I'm using for 63 fires of lung, and to which I am also planning to migrate both Star Story, Dyrt and The Fairy Tale REVAMPED...

**Comment by Tricky1975** [comment]
My friend spoke to me again, and what he told me ain't something you're gonna like
![image](https://user-images.githubusercontent.com/11202073/39326949-d717ccb6-4996-11e8-9a05-9c31c7146b31.png)

Translation:
- Tyrnid: What I am expecting is a buffering issue with his ssd, which is mostly a sign those things are dying
- Tricky: In other words, his hardware is breaking down?
- Tyrnid: That or something I didn't think about, but I expect buffering

I don't know if you have any way to verify his expectations. He knows more about hardware than I do, I can assure you that (which was also the reason I asked him about this).

**Comment by AurelioMV** [comment]
Ok, I'll try the other 2 games tomorrow. I'm pretty sure my SSD ain't dying
hahaha
It's an Samsung EVO 850, which is a very good SSD, and I have the Samsung
Magician installed, it's constantly looking for errors that might occuor in
the drive and it never found any problems.

Either way I'll move it over to the HDD and try running it again!

Em qui, 26 de abr de 2018 16:19, Jeroen Broks <notifications@github.com>
escreveu:
